yyyymmdd/range.py

```diff
@@ -351,9 +382,174 @@ def stop_max_inv(x, y):
             gap = offset - crt
             filler = gap if 0 == gap % step else (gap // step + 1) * step
         start = self.start + crt + filler
-        stop = stop_min(self.stop, other.stop) if sign > 0 else stop_max_inv(self.stop, other.stop)
+        stop = (
+            stop_min(self.stop, other.stop)
+            if sign > 0
+            else stop_max_inv(self.stop, other.stop)
+        )
         return type(self)(start, stop, step)
 
+    # ===========================================================================
+    # mereological part structure
+    # ===========================================================================
+
+    def __lt__(self, other):
```

Comment:
I strongly prefer overriding the order-operators only when an order relation is well-defined and intuitive.
Otherwise, it leads to "surprises", e.g. `a<=b and b<=a` does not imply `a==b`, or `a<b or a>b or a==b is False`.
Please delete the operators.  The `is_proper_part` method suffices.